docs/sql-programming-guide.md

```diff
@@ -733,7 +733,7 @@ SELECT * FROM parquetTable
 
 Table partitioning is a common optimization approach used in systems like Hive. In a partitioned
 table, data are usually stored in different directories, with partitioning column values encoded in
-the path of each partition directory. The Parquet data source is now able to discover and infer
+the path of each partition directory. All built-in data sources are able to discover and infer
```

Comment:
-> `All built-in file sources (including ...)`

You know, it is not applicable to JDBC

Reply:
Oops. Yep. I'll enumerate it.